src/KOKKOS/region_sphere_kokkos.cpp

```diff
@@ -0,0 +1,173 @@
+// clang-format off
+/* ----------------------------------------------------------------------
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   LAMMPS development team: developers@lammps.org
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+/* ----------------------------------------------------------------------
+   Contributing author: Mitch Murphy (alphataubio at gmail)
+------------------------------------------------------------------------- */
+
+#include "region_sphere_kokkos.h"
+
+#include "atom_kokkos.h"
+#include "atom_masks.h"
+
+using namespace LAMMPS_NS;
+
+/* ---------------------------------------------------------------------- */
+
+template<class DeviceType>
+RegSphereKokkos<DeviceType>::RegSphereKokkos(LAMMPS *lmp, int narg, char **arg)
+  : RegSphere(lmp, narg, arg)
+{
+  atomKK = (AtomKokkos*) atom;
+}
+
+/* ----------------------------------------------------------------------
+   inside = 1 if x,y,z is inside or on surface
+   inside = 0 if x,y,z is outside and not on surface
+------------------------------------------------------------------------- */
+
+template<class DeviceType>
+KOKKOS_INLINE_FUNCTION
+int RegSphereKokkos<DeviceType>::k_inside(double x, double y, double z) const
+{
+  const double delx = x - xc;
+  const double dely = y - yc;
+  const double delz = z - zc;
+  const double r = sqrt(delx * delx + dely * dely + delz * delz);
+
+  if (r <= radius) return 1;
+  return 0;
+}
+
+template<class DeviceType>
+void RegSphereKokkos<DeviceType>::match_all_kokkos(int groupbit_in, DAT::tdual_int_1d k_match_in)
+{
+
+  auto execution_space = ExecutionSpaceFromDevice<DeviceType>::space;
+  atomKK->sync(execution_space, X_MASK | MASK_MASK);
+
+  auto d_x = atomKK->k_x.template view<DeviceType>();
+  auto d_mask = atomKK->k_mask.template view<DeviceType>();
+  auto d_match = k_match_in.template view<DeviceType>();
+  auto l_groupbit = groupbit_in;
+
+  copymode = 1;
+
```

Comment:
This is correct use, can remove FIXME comment.

Reply:
i removed the FIXME but left comment to document why useKOKKOS_CLASS_LAMBDA instead of KOKKOS_LAMBDA